File: android/core/src/main/java/com/fjuul/sdk/http/apis/SigningApi.java
package com.fjuul.sdk.http.apis;

import com.fjuul.sdk.entities.SigningKey;

import androidx.annotation.NonNull;
import retrofit2.Call;
import retrofit2.http.GET;

public interface SigningApi {
    @GET("/sdk/signing/v1/issue-key/user")
    @NonNull
    public Call<SigningKey> issueUserKey();
}

File: ios/Core/Sources/Http/Apis/SigningApi.swift
import Foundation
import Alamofire

enum SigningApi: URLRequestConvertible {

    case issueUserKey

    var baseURL: URL {
        return URL(string: "https://dev.api.fjuul.com/sdk/signing/v1")!
    }

    var method: HTTPMethod {
        switch self {
        case .issueUserKey: return .get
        }
    }

    var path: String {
        switch self {
        case .issueUserKey: return "/issue-key/user"
        }
    }

    func asURLRequest() throws -> URLRequest {
        let url = baseURL.appendingPathComponent(path)
        var request = URLRequest(url: url)
        request.method = method
        return request
    }

}
